test/membrane/rtp/session_bin_test.exs

```diff
@@ -27,6 +31,120 @@ defmodule Membrane.RTP.Session.ReceiveBinTest do
   }
 
   @fmt_mapping %{96 => {:H264, 90_000}, 127 => {:MPA, 90_000}}
+  defmodule RTCPPacketInspector do
+    use Membrane.Filter
+
+    @moduledoc """
+    Inspects received RTCP buffers and send notification when all specified RTCP packets were
+    seen at least once.
+    """
+
+    def_input_pad :input, demand_unit: :buffers, caps: :any
+    def_output_pad :output, caps: :any
+
+    def_options expected_sr: [
+                  spec: [list(%{})],
+                  default: [],
+                  description: "List of SSRCs of expected sender reports"
+                ],
+                expected_rr: [
+                  spec: [list(%{})],
+                  defualt: [],
+                  description: "List of SSRCs of expected receiver reports"
+                ]
+
+    @impl true
+    def handle_init(opts) do
+      {:ok, Map.from_struct(opts)}
+    end
+
+    @impl true
+    def handle_demand(:output, size, :buffers, _ctx, state) do
+      {{:ok, demand: {:input, size}}, state}
+    end
+
+    @impl true
+    def handle_process(:input, buffer, _ctx, state) do
+      %Membrane.Buffer{payload: <<head::binary-size(4), body_and_rest::binary>>} = buffer
+      {:ok, %{header: header, length: len}} = Header.parse(head)
+
+      body_size = len - 4
+      <<body::binary-size(body_size), rest::binary>> = body_and_rest
+      {:ok, body} = Packet.parse_body(body, header)
+
+      state =
+        case header.packet_type do
+          201 ->
+            %{
+              state
+              | expected_rr:
+                  Enum.filter(state.expected_rr, &(not compare_on_common_fields(&1, body)))
+            }
+
+          200 ->
+            %{
+              state
+              | expected_sr:
+                  Enum.filter(state.expected_sr, &(not compare_on_common_fields(&1, body)))
+            }
+
+          _ ->
+            state
+        end
+
+      actions = [buffer: {:output, buffer}]
+
+      actions =
+        case Enum.empty?(state.expected_rr) do
+          true ->
+            actions ++ [{:notify, :all_rr_received}]
+
+          _ ->
+            actions
+        end
+
+      actions =
+        case Enum.empty?(state.expected_sr) do
+          true ->
+            actions ++ [{:notify, :all_sr_received}]
+
+          _ ->
+            actions
+        end
+
+      {{:ok, actions}, state}
+    end
+
+    defp compare_on_common_fields(left, right) when is_struct(left) when is_struct(right) do
```

Comment:
This clause makes this function not really comparing on common fields since it ignores the `__struct__` field. Since you seem not to use that anyway, maybe remove this clause?

Reply:
Ok, the reason this clause was introduced because is that it's not allowed to iterate over `{k,v}` pairs in struct, so it was only ensuring that both arguments are maps(which implements Enumerable). Although I agree the action that function takes is not consistent with the naming. Anyway I'm going to completely rewrite it. Thanks.